components/board/fake/board.go

```diff
@@ -370,100 +379,45 @@ func (gp *GPIOPin) SetPWMFreq(ctx context.Context, freqHz uint, extra map[string
+type DigitalInterrupt struct {
+	mu    sync.Mutex
+	conf  board.DigitalInterruptConfig
+	value int64
 }
 
-// NewDigitalInterruptWrapper returns a new digital interrupt to be used for testing.
-func NewDigitalInterruptWrapper(conf board.DigitalInterruptConfig) (*DigitalInterruptWrapper, error) {
-	di, err := pinwrappers.CreateDigitalInterrupt(conf)
-	if err != nil {
-		return nil, err
-	}
-	return &DigitalInterruptWrapper{
-		di:        di,
-		callbacks: map[chan board.Tick]struct{}{},
-		conf:      conf,
+// NewDigitalInterrupt returns a new digital interrupt to be used for testing.
+func NewDigitalInterrupt(conf board.DigitalInterruptConfig) (*DigitalInterrupt, error) {
+	return &DigitalInterrupt{
+		conf: conf,
 	}, nil
 }
 
-func (s *DigitalInterruptWrapper) reset(conf board.DigitalInterruptConfig) error {
+func (s *DigitalInterrupt) reset(conf board.DigitalInterruptConfig) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	reconf, isReconf := s.di.(pinwrappers.ReconfigurableDigitalInterrupt)
-	if conf.Name != s.conf.Name || !isReconf {
-		// rebuild
-		di, err := pinwrappers.CreateDigitalInterrupt(conf)
-		if err != nil {
-			return err
-		}
-		s.conf = conf
-		s.di = di
-		for c := range s.callbacks {
-			pinwrappers.AddCallback(di.(*pinwrappers.BasicDigitalInterrupt), c)
-		}
-		return nil
-	}
-	// reconf
-	if err := reconf.Reconfigure(conf); err != nil {
-		return err
-	}
 	s.conf = conf
-	return nil
+}
+
+// RemoveCallback removes a listener for interrupts.
+func (s *DigitalInterrupt) RemoveCallback(c chan board.Tick) {
 }
 
 // Value returns the current value of the interrupt which is
 // based on the type of interrupt.
-func (s *DigitalInterruptWrapper) Value(ctx context.Context, extra map[string]interface{}) (int64, error) {
+func (s *DigitalInterrupt) Value(ctx context.Context, extra map[string]interface{}) (int64, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	if s.conf.Pin == nonZeroInterruptPin {
 		s.value++
 		return s.value, nil
 	}
-	return s.di.Value(ctx, extra)
-}
-
-// Tick is to be called either manually if the interrupt is a proxy to some real
-// hardware interrupt or for tests.
-// nanoseconds is from an arbitrary point in time, but always increasing and always needs
-// to be accurate.
-func Tick(ctx context.Context, s *DigitalInterruptWrapper, high bool, nanoseconds uint64) error {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	return pinwrappers.Tick(ctx, s.di.(*pinwrappers.BasicDigitalInterrupt), high, nanoseconds)
-}
-
-// AddCallback adds a callback to be sent a low/high value to when a tick
-// happens.
-func AddCallback(s *DigitalInterruptWrapper, c chan board.Tick) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	s.callbacks[c] = struct{}{}
-	pinwrappers.AddCallback(s.di.(*pinwrappers.BasicDigitalInterrupt), c)
-}
-
-// RemoveCallback removes a listener for interrupts.
-func (s *DigitalInterruptWrapper) RemoveCallback(c chan board.Tick) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	delete(s.callbacks, c)
-	s.di.RemoveCallback(c)
+	return 0, nil
```

Comment:
This seems like a change in behavior: we used to return the value currently stored in the interrupt, and now we return 0 no matter what. Why do that? 

edit: it's not as big a change as I thought: there wasn't a way to update the value, so returning the value _would_ always return 0. I wonder if the original intention was that there would be a way to set the fake value and then we'd always return whatever it was currently set to, but we didn't make that first part.

Personally, I find this behavior confusing, that one of the pins increments the value every time you check it and all the others are 0 forever. I don't understand why there are two implementations like that. Now that we're using this in fewer tests, could all the pins auto-increment (or could all of them be 0 forever)?

Reply:
Yeah it still has the same behavior it did with the pinwrapper. I think @martha-johnston made this change recently? Don't have a strong opinion on how the fake interrupt should work but I can make a ticket to change it in a different PR.

Reply:
Yeah Alan this was the PR of mine I believe you reviewed to have non-zero values returned from fake board analogs and digital interrupts to see the behavior on the RC card, so this one just increments (rather than returning a random value so it's more similar to how digital interrupts actually work) on a specific `nonZeroInterruptPin`. there's a comment explaining it all at the top of the file where that variable is declared, but it basically says it will continue to return 0 so we can use that function for testing, but if you use pin 4 iirc, you will get this incrementing value so you can see it changing on the RC card

Reply:
well, before it would return `s.di.Value`, but that was always 0 anyways so this change Olivia has made should not change the behavior

Reply:
:shrug: Leave the PR as-is, I guess.